internal/apiresource/pipeline.go

```diff
@@ -0,0 +1,162 @@
+				branchName = defaultGitRepoBranch
+			}
+
+			cloneTask := v1beta1.PipelineTask{
+				Name:    cloneTaskName,
+				TaskRef: &v1beta1.TaskRef{Name: "git-clone"},
+				Workspaces: []v1beta1.WorkspacePipelineTaskBinding{
+					v1beta1.WorkspacePipelineTaskBinding{Name: "output", Workspace: irpipeline.WorkspaceName},
+				},
+				Params: []v1beta1.Param{
+					v1beta1.Param{Name: "url", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: gitRepoURL}},
+					v1beta1.Param{Name: "revision", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: branchName}},
+					v1beta1.Param{Name: "deleteExisting", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: "true"}},
+				},
+			}
+			if !firstTask {
+				cloneTask.RunAfter = []string{prevTaskName}
+			}
+
+			imageName := container.ImageNames[0]
+			// Assume there is no git repo. If there is no git repo we can't do CI/CD.
+			dockerfilePath := dockerfilePathPlaceholder
+			contextPath := contextPathPlaceholder
+			// If there is a git repo, set the correct context and dockerfile paths.
+			if container.RepoInfo.GitRepoDir != "" {
+				relDockerfilePath, err := filepath.Rel(container.RepoInfo.GitRepoDir, container.RepoInfo.TargetPath)
+				if err != nil {
+					log.Errorf("Failed to make the path %q relative to the path %q Error %q", container.RepoInfo.GitRepoDir, container.RepoInfo.TargetPath, err)
+				} else {
+					dockerfilePath = relDockerfilePath
+					// We can't figure out the context from the source. So assume the context is the directory containing the dockerfile.
+					contextPath = filepath.Dir(relDockerfilePath)
+				}
+			}
+
+			buildPushTaskName := "build-push-" + fmt.Sprint(i)
+			buildPushTask := v1beta1.PipelineTask{
+				RunAfter: []string{cloneTaskName},
+				Name:     buildPushTaskName,
+				TaskRef:  &v1beta1.TaskRef{Name: "kaniko"},
+				Workspaces: []v1beta1.WorkspacePipelineTaskBinding{
+					v1beta1.WorkspacePipelineTaskBinding{Name: "source", Workspace: irpipeline.WorkspaceName},
+				},
+				Params: []v1beta1.Param{
+					v1beta1.Param{Name: "IMAGE", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: "$(params.image-registry-url)/" + imageName}},
+					v1beta1.Param{Name: "DOCKERFILE", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: dockerfilePath}},
+					v1beta1.Param{Name: "CONTEXT", Value: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeString, StringVal: contextPath}},
+				},
+			}
+			tasks = append(tasks, cloneTask, buildPushTask)
+			firstTask = false
+			prevTaskName = buildPushTaskName
+		} else if container.ContainerBuildType == plantypes.S2IContainerBuildTypeValue {
+			log.Errorf("TODO: Implement support for S2I")
+		} else if container.ContainerBuildType == plantypes.CNBContainerBuildTypeValue {
+			log.Errorf("TODO: Implement support for CNB")
+		} else {
+			log.Errorf("Unknown containerization method: %v", container.ContainerBuildType)
+		}
```

Comment:
Lets make these as warnings rather than error. we have to think of supporting the reusedockerfile usecase too.

Reply:
I thought error is better since this is something that is to be fixed.
Warning is something that may not need to be fixed.